apps/src/code-studio/components/ExportAllowedDialog.jsx

```diff
@@ -0,0 +1,744 @@
+              id="radioAndroid"
+              readOnly
+              checked
+            />
+            <label htmlFor="radioAndroid" style={styles.radioLabel}>
+              I have an Android device
+            </label>
+          </div>
+          <div>
+            <input
+              style={styles.radioInput}
+              type="radio"
+              id="radioIOS"
+              disabled
+            />
+            <label htmlFor="radioIOS" style={styles.radioLabelDisabled}>
+              I have an iOS device (currently not supported)
+            </label>
+          </div>
+        </div>
+        <div style={styles.section}>
+          <p style={styles.p}>
+            <b>Note: </b>Exporting will take 10-15 minutes. If you change your
+            app after you start exporting, those changes will not be included.
+          </p>
+        </div>
+      </div>
+    );
+  }
+
+  renderIconPage() {
+    return (
+      <div>
+        <div style={styles.section}>
+          <p style={styles.title}>Upload your App icon</p>
+        </div>
+        <div style={styles.section}>
+          <div style={styles.icon}>
+            <img style={styles.iconImage} src={exportExpoIconPng} />
+          </div>
+          <button
+            type="button"
+            style={styles.uploadIconButton}
+            onClick={this.onUploadIcon}
+          >
+            Upload another image
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  renderPublishPage() {
+    return (
+      <div>
+        <div style={styles.section}>
+          <p style={styles.title}>Create Android Package</p>
+        </div>
+        <div style={styles.section}>
+          <p style={styles.p}>
+            An Android Package (APK) is a package of code and other files that
+            can be installed as an app on an Android device.
+          </p>
+          <p style={styles.p}>
+            <b>Note: </b>After you click "Create", it will take about 10-15
+            minutes to create the package.
+          </p>
+        </div>
+      </div>
+    );
+  }
+
+  renderGeneratingPage() {
+    const {
+      exporting,
+      generatingApk,
+      showSendToPhone,
+      exportError,
+      apkError,
+      apkUri = ''
+    } = this.state;
+    const waiting = exporting || generatingApk;
+    const error = exportError || apkError;
+    const {appType, channelId} = this.props;
+    const titleText = waiting
+      ? 'Creating Android Package...'
+      : error
+      ? 'Error creating Android Package'
+      : 'The Android Package was created successfully';
+    const headerText = waiting
+      ? 'Please wait for about <b>15 minutes</b>.'
+      : error || 'Send this link to your device to install the app.';
+    return (
+      <div>
+        <div style={styles.section}>
+          <p style={styles.title}>{titleText}</p>
+        </div>
+        <div style={styles.section}>
+          <p style={styles.p} dangerouslySetInnerHTML={{__html: headerText}} />
```

Comment:
prefer using the [UnsafeRenderedMarkdown](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/UnsafeRenderedMarkdown.jsx) component when we want to format strings

Reply:
I just played with it now and it isn't really working. I can't put it inside a `<p>` tag. When I put it inside a `<div>`, it loses the styles I applied. Plus it instantiates a markdown processor that I don't need. Seems like a good idea to have a reusable component, but it doesn't seem like the right fit for this scenario.